### PR TITLE
Added support for areaTypesPatents

### PR DESCRIPTION
**Pull request by j-s-rawat**
### What

Introduced API client function to support getting parents for areaTypes

### How to review

- All test pass 
- The response matches the end-point in dp-population-types-api

### Who can review

Anyone

### CODE REVIEW

**Review by nimitsarup — DISMISSED** [review]

File: population/area_types.go [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-// AreaType is an area-type model with ID and Label
+// AreaType is an area type model with ID and Label
 type AreaType struct {
 	ID         string `json:"id"`
 	Label      string `json:"label"`
 	TotalCount int    `json:"total_count"`
 }
 
-// GetAreaTypesResponse is the response object for GET /area-types
+// GetAreaTypesResponse is the response object for GET /area_types
```
Comment (ChrysmOre) [inline population/area_types.go:10]: URLs are `camel-case`, JSON is `snake_case`

File: population/areas.go [hunk 2]
```diff
@@ -4,10 +4,22 @@ package population
...
 // GetAreasResponse is the response object for GET /areas
 type GetAreasResponse struct {
 	Areas []Area `json:"areas"`
 }
+
+// Area is an area model with ID and Label
+type AreaTypes struct {
+	ID         string `json:"id"`
+	Label      string `json:"label"`
+	TotalCount string `json:"total-count"`
```
Comment (ChrysmOre) [inline population/areas.go:19]: you probable wanted to change to `total_count` here

File: population/client_test.go [hunk 3]
```diff
@@ -530,6 +530,171 @@ func TestGetPopulationTypes(t *testing.T) {
...
+			UserAuthToken:    userAuthToken,
+			ServiceAuthToken: serviceAuthToken,
+		}
+		_, err := client.GetAreaTypesParents(context.Background(), input)
+
+		Convey("it should return an internal error", func() {
+			So(err, shouldBeDPError, http.StatusInternalServerError)
+		})
+	})
+	/* Currently the swagger spec in dp-population-type-api doesn't expect any error codes
```
Comment (ChrysmOre) [inline population/client_test.go:618]: You should probably uncomment these tests or remove them. The only hard-coded status code in the endpoint is 500 but any status code returned from Cantabular can also be returned. And there are many reasons a status code could be returned other than what is defined in the swagger spec, if Cantabular gets updated or the service is unreachable etc

File: population/areas.go (hunk 2 above)
Comment (ChrysmOre) [reply population/areas.go:19]: Also shouldn't AreaTypes be in area_type.go? Same for GetAreaTypesParentsResponse. Actually we should have a contract.go for that..

**Review by PauloPortugal — APPROVED** [review]
:+1:

**Pull request merged**